### PR TITLE
Make app more Android Dagger2 friendly

### PR DESCRIPTION
**Pull request by danielpassos**
This PR update the app to Android 26 and make this more Android Dagger2 friendly.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -33,56 +33,62 @@ android {
...
+        exclude group: "com.android.support", module: "support-annotations"
+        exclude group: "com.android.support", module: "appcompat-v7"
+        exclude group: "com.android.support", module: "support-v4"
+        exclude group: "com.android.support", module: "design"
+        exclude module: "recyclerview-v7"
     }
-    androidTestAnnotationProcessor 'com.google.dagger:dagger-compiler:2.11'
-    androidTestAnnotationProcessor 'com.google.dagger:dagger-android-processor:2.11'
+    androidTestAnnotationProcessor "com.google.dagger:dagger-compiler:2.11"
+    androidTestAnnotationProcessor "com.google.dagger:dagger-android-processor:2.11"
```
Comment (wei-lee) [inline app/build.gradle:93]: is there a big difference between single quote and double quote in gradle?

**Comment by wei-lee** [comment]
@danielpassos Thanks for the contribution, this is great. We can merge this once the build is fixed. Looks like there are some tests failing.

Is there a good document on how to use Dagger? I find the official doc is bit confusing.

File: app/build.gradle (hunk 1 above)
Comment (danielpassos) [reply app/build.gradle:93]: Single quote don't allow variable `"com.google.dagger:dagger-compiler:$some_variable"`

**Comment by danielpassos** [comment]
Well, I was about to comment that in the meeting, Dagger (also all Android) documentation is a hell. So, there is a [documentation about Android stuffs](https://google.github.io/dagger/android.html#base-framework-types) but it's not too much clear and the example are confused like usually.

**Comment by wei-lee** [comment]
@danielpassos have you come across any better documents explain how to use Dagger? There are a few on Dagger 1, but not too much on Dagger 2.

**Comment by danielpassos** [comment]
@wei-lee Well, there is no good documentation about Dagger 2 IMO. All my knowledge came from doing real app and stealing idea from the other open source projects.

**Comment by danielpassos** [comment]
Closing for now, since I'm having problem to fix the tests in this way

**Pull request closed without merging**